### ISSUE TITLE
Vanilla JS option

### ISSUE THREAD

**Issue by pixelbucket-dev** [ionic-team/app-wizard-issues#6]
**Describe the bug**
Have a third (fourth) option "Javascript". This could behave similar for Stencil PWA toolkit but with extra goodies of a "proper" Ionic project and Ionic CLI.

**Additional context**
Choices of React OR Angular are too narrow. I assume an addition for vanilla JS would not be too difficult. You could use the PWA toolkit to create a Stencil App with Ionic but then you don't get all the other goodies like the integrated CLI and scripts.

**Comment by mlynch** [comment]
Thanks! Definitely something we've been talking through. Will keep this issue open for feedback.

**Comment by pixelbucket-dev** [comment]
Thanks.

Interestingly, with the cli `ionic init` you can select a third option `custom`, which basically allows to convert any app, even an Ionic PWA toolkit (Stencil App) into an Ionic app, there are a few extra steps involved, but the CLI gives the right hints 👍. 

However, with `ionic start` that option does not exit. Therefore, rather than having to go the detour with scaffolding a vamilla JS project manually and then running `ionic init` on it just doing a simple `ionic start` would be perfect :).

**Comment by BerkeAras** [comment]
There is already an open issue [#4390](https://github.com/ionic-team/ionic-cli/issues/4390) for the CLI to add vanilla JS option. There are many developers who are using ionic without or with another framework, and it is bad, just to include it everytime using the cdn, or create a capacitor project, add ionic, it takes time.
I'm just waiting for Ionic to just add vanilla js to the CLI and the App Wizard because it is the most important part of ionic (angular, react, vue are just built on core version)

**Comment by pixelbucket-dev** [comment]
@BerkeAras I agree. Especially since the documentation for the components has an option to select _JavaScript_ (besides _React_ and _Angular_). There is a clear disconnet between this and what the CLI offers. I actually always wondered why this exists because _JavaScript_ is neither advertised in the starting guide and neither with the "framework" options in the documentation menu (it has even the _Vue_ (beta) entry, which is another inconsistency).

**Comment by BerkeAras** [comment]
Why isn't here any updates? Why is ionic not adding Vanilla JS to the app wizard and the cli? I cannot understand...

**Comment by pixelbucket-dev** [comment]
@BerkeAras they probably have a lot on their plate, so you have to patient. If you need that quicker, maybe you can help out with the pull request?

**Comment by BerkeAras** [comment]
@mlynch Any updates?

**Comment by mlynch** [comment]
One reason that by giving Vanilla JS as an official option, we need to make sure the whole experience and toolchain brings vanilla up to the level of React, Angular, and (soon) Vue. That's just not going to happen overnight.

Of course, you're very welcome to use Ionic in a vanilla set up (we do that here in a number of cases), but the toolchain/routing/etc. around it isn't as mature.

**Comment by reesericci** [comment]
its a year later, any updates?

**Comment by ahait** [comment]
It's now 2022 almost summer....  a vanillaJS "type" would really be awesome

**Comment by CrookedGrin** [comment]
I am also interested in importing Ionic web components directly via an import statement rather than pointing to the CDN, so that I don't have to rely on the CDN loading everything at runtime. If I'm already installing `@ionic/core`, is there a way to point to specific component packages and have them be included? We're using Webpack for bundling, but no frontend framework because it's a Rails app, which has its own idiomatic JS stuff (Hotwire/Turbo).

**Comment by ghost** [comment]
This is going to be very useful

**Comment by CommanderGL** [comment]
3 Years still not added.

**Comment by surfjedi** [comment]
please add it!

**Comment by Sneffel** [comment]
We are one step closer to being free as humankind... Please 🙏🏿

**Comment by dustinchen93** [comment]
2025，Any progressions?
